constraint/pkg/client/client.go

```diff
@@ -93,7 +123,9 @@ func (c *Client) AddTemplate(ctx context.Context, templ *templates.ConstraintTem
 		}
 	}
 
-	if cachedCpy != nil && cachedCpy.SemanticEqual(templ) {
+	// if there is more than one active driver for the template, there is some cleanup to do
+	// from a botched driver swap.
+	if cachedCpy != nil && cachedCpy.SemanticEqual(templ) && len(cached.activeDrivers) == 1 {
```

Comment:
why is this only true when there is only 1 activeDrivers?

Reply:
If the contents haven't change and there is only one active driver, there is no work to do and we can return early.

If the contents have changed, then there is work to do to update the template code.

If the contents have not change, but there is more than one active driver, then there is cleanup to do (only one driver should be active at a time).

If the contents have not changed, and there are zero active drivers... this state shouldn't happen, but we should add the template code to a driver so we have an active driver.